app/screens/AppNavigator/screens/Dex/PoolSwap/PoolSwapScreen.tsx

```diff
@@ -0,0 +1,209 @@
+import { PrimaryButton } from '../../../../../components/PrimaryButton'
+import { PrimaryColor, PrimaryColorStyle } from '../../../../../constants/Theme'
+import { useTokensAPI } from '../../../../../hooks/wallet/TokensAPI'
+import { translate } from '../../../../../translations'
+import LoadingScreen from '../../../../LoadingNavigator/LoadingScreen'
+import { DexParamList } from '../DexNavigator'
+
+interface TokenForm {
+  control: Control
+  controlName: string
+  token: AddressToken
+  onMaxPress?: (amount: string) => void
+  title: string
+}
+
+interface SwapSummaryItems {
+  poolpair: PoolPairData
+  tokenA: AddressToken
+  tokenB: AddressToken
+}
+
+type Props = StackScreenProps<DexParamList, 'PoolSwapScreen'>
+
+export function PoolSwapScreen ({ route }: Props): JSX.Element {
+  const poolpair = route.params.poolpair
+  const tokens = useTokensAPI()
+  const [tokenAForm, tokenBForm] = ['tokenA', 'tokenB']
+  const [tokenA, setTokenA] = useState<AddressToken>()
+  const [tokenB, setTokenB] = useState<AddressToken>()
+  const { control, setValue, formState: { isValid }, getValues, trigger } = useForm({ mode: 'onChange' })
+
+  function onSubmit (): void {
+    console.log(isValid)
+    console.log(getValues())
+  }
+
+  function swapToken (): void {
+    setTokenA(tokenB)
+    setTokenB(tokenA)
+  }
+
+  useEffect(() => {
+    const a = tokens.find((token) => token.id === poolpair.tokenA.id)
+    setTokenA(a)
+    const b = tokens.find((token) => token.id === poolpair.tokenB.id)
+    setTokenB(b)
+  }, [])
+
+  if (tokenA === undefined || tokenB === undefined) {
+    return <LoadingScreen />
+  }
+
+  return (
+    <ScrollView style={tailwind('bg-gray-100')}>
+      <TokenRow
+        token={tokenA} control={control} controlName={tokenAForm}
+        title={translate('screens/PoolSwapScreen', 'From')} onMaxPress={async (amount) => {
+          setValue(tokenAForm, amount)
+          await trigger(tokenAForm)
+        }}
+      />
+      <TouchableOpacity style={tailwind('justify-center items-center mt-4')} onPress={swapToken} testID='swap_button'>
+        <MaterialIcons name='swap-vert' size={28} color={PrimaryColor} />
+      </TouchableOpacity>
+      <TokenRow
+        token={tokenB} control={control} controlName={tokenBForm}
+        title={translate('screens/PoolSwapScreen', 'To')}
+      />
+      {
+        (new BigNumber(getValues()[tokenAForm]).isGreaterThan(0) && new BigNumber(getValues()[tokenBForm]).isGreaterThan(0)) &&
+          <SwapSummary poolpair={poolpair} tokenA={tokenA} tokenB={tokenB} />
+      }
+      <PrimaryButton disabled={!isValid} title='Swap' onPress={onSubmit} testID='button_submit'>
+        <Text style={tailwind('text-white font-bold')}>{translate('screens/PoolSwapScreen', 'SWAP')}</Text>
+      </PrimaryButton>
+    </ScrollView>
+  )
+}
+
+function TokenRow ({ token, control, onMaxPress, title, controlName }: TokenForm): JSX.Element {
```

Comment:
you should put TokenForm interface near the JSX.Element function, easier to read code when reviewing